lib/core/src/display/navigation/camera_manager.rs

```diff
@@ -0,0 +1,197 @@
+        let amount = event.position.y - event.previous_position.y;
+        Self { focus, amount }
+    }
+}
+
+// ====================
+// === PanningEvent ===
+// ====================
+
+/// A struct holding panning event information.
+pub struct PanningEvent {
+    pub movement : Vector2<f32>
+}
+
+impl PanningEvent {
+    fn from(event:MousePositionEvent) -> Self {
+        let mut movement   = event.position - event.previous_position;
+                movement.x = -movement.x;
+        Self { movement }
+    }
+}
+
+// =============
+// === Event ===
+// =============
+
+/// An enumeration representing navigation events.
+pub enum Event {
+    Zooming(ZoomingEvent),
+    Panning(PanningEvent)
+}
+
+// ====================
+// === MovementType ===
+// ====================
+
+#[derive(PartialEq)]
+enum MovementType {
+    Panning,
+    Zooming { focus : Vector2<f32> }
+}
+
+// ========================
+// === EventHandlerData ===
+// ========================
+
+struct CameraManagerData {
+    event_queue       : Vec<Event>,
+    movement_type     : Option<MovementType>,
+    mouse_position    : Vector2<f32>
+}
+
+impl Default for CameraManagerData {
+    fn default() -> Self {
+        let event_queue       = default();
+        let mouse_position    = zero();
+        let movement_type     = None;
+        Self { event_queue, mouse_position, movement_type }
+    }
+}
+
+// =====================
+// === CameraManager ===
+// =====================
+
+/// Struct used to handle mouse events such as MouseDown, MouseMove, MouseUp,
+/// MouseLeave and MouseWheel.
+pub struct CameraManager {
+    mouse_manager : MouseManager,
+    data          : Rc<RefCell<CameraManagerData>>
+}
+
+impl CameraManager {
+    pub fn new(dom:&DOMContainer) -> Result<Self> {
+        let mouse_manager     = MouseManager::new(dom)?;
+        let data              = default();
+        let data              = Rc::new(RefCell::new(data));
+        let mut event_handler = Self { mouse_manager, data };
+
+        event_handler.initialize_events()?;
+        Ok(event_handler)
+    }
+
+    /// Checks if EventHandler has an event.
+    pub fn poll(&mut self) -> Option<Event> {
+        self.data.borrow_mut().event_queue.pop()
+    }
+
+    // Initialize mouse events.
+    fn initialize_events(&mut self) -> Result<()> {
+        self.disable_context_menu()?;
+        self.initialize_wheel_zooming()?;
+        self.initialize_interaction_start_event()?;
+        self.initialize_mouse_move_event()?;
+        self.initialize_interaction_end_event()
+    }
+
+    fn initialize_wheel_zooming(&mut self) -> Result<()> {
+        let data = self.data.clone();
+        self.mouse_manager.add_mouse_wheel_callback(move |event:MouseWheelEvent| {
+            let zoom_event = ZoomingEvent::from_mouse_wheel(event, &data);
+            let event      = Event::Zooming(zoom_event);
+            data.borrow_mut().event_queue.push(event);
+        })?.forget();
```

Comment:
We should rather not use `forget` so often. What happens if we would like to remove this callback in the future and change mouse to joystick or something?